src/schema/types/user.graphql

```diff
@@ -14,12 +14,26 @@ input UserCreationInput {
   password: String!
 }
 
+input SignupInput {
+  firstName: String!
+  lastName: String!
+  username: String!
+  password: String!
+}
+
+type AuthenticationResult {
+  user: User
+  jwt: String
+  authError: String
+}
```

Comment:
Error reporting is built-in on GraphQL (see my comment above). This type should be like this:
```graphql
type AuthenticationResult {
  user: User!
  jwt: String!
}
```